modules/core/include/opencv2/core.hpp

```diff
@@ -150,6 +150,14 @@ It is possible to alternate error processing by using #redirectError().
  */
 CV_EXPORTS CV_NORETURN void error(const Exception& exc);
 
+/*! @brief Flags that allow to midify some functions behavior. Used as set of flags.
+*/
+enum ImplementationHint {
+    IMPL_DEFAULT = 0, //!< Use generic portable implementation
+    IMPL_ALLOW_APPROXIMATION = 1, //!< Allow alternative approximations to get faster implementation. Behaviour and result depends on a platform
+    IMPL_INHERIT = 1 << 31, //!< Promote the hints to internal subroutines
+};
```

Comment:
'Implementation' word is not good, in my opinion, too long to type.
`AlgoHint` is better.

```
enum class Algo {
    Default = 0,
    Fast = 1,
    Inherit = 2
};
```

Reply:
Renamed the enum and removed hint for now as we do not have a case for it.